File: programmers/Lv_1/KimHyunseo/소수찾기.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int solution(int n) {
	int answer = 0;
	vector<bool> check(n + 1, true);
	for (int i = 2; i <=n; i++)
	{
		if (check[i])
			answer +=1;
		for (int j = i; j <=n; j+=i)
		{
			check[j] = false;
		}
	}
	return answer;
}

int main()
{
	int n = 10;
	solution(10);
}

File: programmers/Lv_1/HwangJunghoon/소수 찾기.py
def solution(n):

    """
    0 : 소수가 아님, 1 : 소수
    [0,0] : 0과 1은 소수가 아니므로 0 표시
    [1] * (n-1) : 2부터 n까지 (n-1) 소수라고 가정하고 시작
    """
    answer = [0, 0] + [1] * (n - 1)
    # int(n**0.5) : 2부터 n까지 검사할 필요 없이 제곱근까지만 검사
    for i in range(2, int(n**0.5)+1):
        if answer[i]:
            #소수를 제외하고 그 소수의 배수만 0으로 치환, (n-i)//i : 소수의 배수의 갯수
            answer[i*2::i] = [0] * ((n - i) // i)
    # 1로 남은 것만 갯수를 세어 return
    return answer.count(1)
print(solution(1000000))